go/libraries/doltcore/env/actions/commit.go

```diff
@@ -169,10 +169,16 @@ func CommitStaged(ctx context.Context, dbData env.DbData, props CommitStagedProp
 
 	// DoltDB resolves the current working branch head ref to provide a parent commit.
 	// Any commit specs in mergeCmSpec are also resolved and added.
-	c, err := ddb.CommitWithParentSpecs(ctx, h, rsr.CWBHeadRef(), mergeCmSpec, meta)
+	c, err := ddb.CommitWithParentSpecs(ctx, h, rsr.CWBHeadRef(), mergeCmSpec, meta)z
 
-	if err == nil {
-		err = rsw.ClearMerge()
+	if err != nil {
+		return "", err
+	}
+
+	err = rsw.ClearMerge()
+
+	if err != nil {
+		return "", err
 	}
 
 	h, err = c.HashOf()
```

Comment:
Here too, right? `h` could be `nil` if `c.HashOf()` returns `err != nil`?

Reply:
nah, can't be nil... but it's more idiomatic.